chapters/traversal.md

```diff
@@ -1,83 +1,89 @@
 
-A rooted recursive representation has a root node and every node has X children.
-In Jina, the root node is the document itself, while the *left* & *right* children are referred to as *chunks* and *matches* respectively.
+## General overview
+
+In Jina, the term Document refers to more than just a simple dataset. A Document is used for both indexing and querying. A search query is also a Document itself.
+
+When you index a Document in Jina, it can be represented as a rooted recursive representation (tree). The representation consists of a *root node* and several *child nodes*. In Jina, the root node is the original Document itself, while the *left* and *right* children are referred to as *chunks* and *matches* respectively. The figure below illustrates a basic Document structure that consists of a Document (root node) as well as two child nodes (chunks and matches).
 
 ![rooted-binary-tree](./images/overview.png)
 
-The above image illustrates a basic document structure: A document (root node) and the two possible child nodes: *chunks* and *matches*.
-chunks are a sequence of documents which is attached to the root document with a higher `granularity` degree. `matches` is a sequence of documents which are semantically related to the root document.
-We'll dive into these concepts in this chapter:
+The two terms chunks and sequence are still a bit unclear. In short, chunks are a sequence of Documents which can be attached to any Document with a higher `granularity` degree. In the figure above this is the root node. 
+
+Similarly, this applies to `matches`. `matches` is a sequence of Documents which are semantically related to the higer-level Document (the root node in the figure above). We will dive into these concepts in more detail below.
 
 - [Chunks](#chunks)
 - [Matches](#matches)
 - [Let's go deeper: Recursive Document Representation](#lets-go-deeper-recursive-document-representation)
 - [Document Traversal with traversal paths](#document-traversal-with-traversal-paths)
 
-
 ## Chunks
 
-Each Jina Document (potentially) consists of a list of chunks. A `chunk` is a small semantic unit of a Document, like a sentence or a 64x64 pixel image patch.
+### Understanding chunks
+
+Each Jina Document (potentially) consists of a list of *chunks*. A single chunk is a small semantic unit of a Document, like a sentence or a 64x64 pixel image patch.
 
-Think about these use cases: you want to search a document at a specific `granularity` level, e.g. a sentence or a paragraph.
-Or your query consists of multiple modalities, such as a piece of text together with an image.
-`chunk` makes it feasible!
+In practice, chunks correspond to a certain level of granularity, for example a paragraph or a sentence. A chunk also allows you to search for a section of the Document, for example an image followed by a paragraph of text. These are use cases that are more common than you might think.
```

Comment:
Write about the identifier for chunks `chunk_id` : what kind of ids are assigned for a document that has say an image, text and video (3 different modalities) in sequence (eg occurence in PDF)

Reply:
I have added that in general, and will go through the docs to figure out the different types of ids, now